scio-core/src/main/scala/com/spotify/scio/io/dynamic/package.scala

```diff
@@ -122,6 +129,34 @@ package object dynamic {
 
   }
 
+  implicit class DynamicProtobufSCollection[T <: Message](private val self: SCollection[T])
+      extends AnyVal {
+
+    def saveAsDynamicProtobufFile(
+      path: String,
+      numShards: Int = 0,
+      suffix: String = ".protobuf",
+      codec: CodecFactory = CodecFactory.deflateCodec(6),
+      metadata: Map[String, AnyRef] = Map.empty
+    )(destinationFn: T => String)(implicit cd: ClassTag[T], coder: ScioCoder[T]): Future[Tap[T]] = {
+      val elemCoder = CoderMaterializer.beam(
+        self.context,
+        coder
+      )
+      self.saveAsDynamicAvroFile(
+        path,
+        numShards,
+        AvroBytesUtil.schema,
+        suffix,
+        codec,
+        metadata + ("protobuf.generic.schema" -> AvroBytesUtil.schema.toString)
```

Comment:
@xBATx you were get serde issues because avro `Schema` is not serializable. Doing a `toString` did the trick. However for consistency we need it in `json` format.

```scala
import me.lyh.protobuf.generic
val schemaJson = generic.Schema.of[Message](ct.asInstanceOf[ClassTag[Message]]).toJson
```